### PR TITLE
doas 6.3p12 (new formula)

### PR DESCRIPTION
**Pull request by Neved4**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Related: https://github.com/Homebrew/homebrew-core/pull/70073

### CODE REVIEW

**Review by alerque — COMMENTED** [review]
Have you seen #70073?

File: Formula/d/doas.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  sha256 "e4f37745345c12d4e0c8c03c8237791729cf047dbd7b2455f8de60e2f82ac1b0"
+  license "BSD-2-Clause"
+  head "https://github.com/slicer69/doas.git", branch: "master"
+
+  bottle do
+    root_url "https://github.com/Neved4/homebrew-tap/releases/download/doas-6.3p12"
+    sha256 cellar: :any_skip_relocation, arm64_sonoma: "3973d3f674d9de9fe6402e108064524105f82af8cf464ba97f22cb1a73742a03"
+    sha256 cellar: :any_skip_relocation, ventura:      "2addf400597918085b22645fb17d45c9cd420298a9cdb60a41540d7d6c4c6eb8"
+    sha256 cellar: :any_skip_relocation, monterey:     "a922ab3545bb5714dc994b77a4e81e6653422f34de7072cce2d78203cbd99c34"
+  end
```
Comment (carlocab) [inline Formula/d/doas.rb:15]:
```suggestion
```

File: Formula/d/doas.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+  def install
+    system "make", "all"
+    bin.install "doas"
+    bin.install "doasedit"
+    bin.install "vidoas"
+    man1.install "doas.1"
+    man5.install "doas.conf.5"
+    man8.install "doasedit.8"
+    man8.install "vidoas.8"
```
Comment (carlocab) [inline Formula/d/doas.rb:27]:
```suggestion
    system "make", "PREFIX=#{prefix}", "SYSCONFDIR=#{etc}", "install"
```

File: Formula/d/doas.rb [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  head "https://github.com/slicer69/doas.git", branch: "master"
+
+  bottle do
+    root_url "https://github.com/Neved4/homebrew-tap/releases/download/doas-6.3p12"
+    sha256 cellar: :any_skip_relocation, arm64_sonoma: "3973d3f674d9de9fe6402e108064524105f82af8cf464ba97f22cb1a73742a03"
+    sha256 cellar: :any_skip_relocation, ventura:      "2addf400597918085b22645fb17d45c9cd420298a9cdb60a41540d7d6c4c6eb8"
+    sha256 cellar: :any_skip_relocation, monterey:     "a922ab3545bb5714dc994b77a4e81e6653422f34de7072cce2d78203cbd99c34"
+  end
+
+  depends_on :macos
```
Comment (carlocab) [inline Formula/d/doas.rb:17]:
```suggestion
```
I don't think this is accurate, since `doas` supports Linux.

**Review by carlocab — COMMENTED** [review]
Also, does this not require installation as setuid root?

File: Formula/d/doas.rb [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    man8.install "doasedit.8"
+    man8.install "vidoas.8"
+  end
+
+  def caveats
+    <<~EOS
+      To complete the installation, run the following command manually:
+        sudo cp /etc/pam.d/sudo /etc/pam.d/doas
+    EOS
+  end
```
[the remark this answers is not in the document]
Comment (Neved4) [reply Formula/d/doas.rb:35]: It's true! It's already in upstream docs and not Homebrew specific. I thought it was good to guide users to the additional steps explicitly. Would there be a better place to indicate it?

**Comment by Neved4** [comment]
@carlocab Thx for the suggestions! I was aware of some of these, my PR was a WIP draft. I appreciate it a lot!

Yes, it does require setuid root but not sure how to achieve it via `brew`.

Now that I'm looking this functionality has been declined multiple times: https://github.com/Homebrew/homebrew-core/pull/14428
https://github.com/Homebrew/legacy-homebrew/issues/43862
https://github.com/Homebrew/legacy-homebrew/issues/40992

File: Formula/d/doas.rb (hunk 4 above)
[the remark this answers is not in the document]
Comment (carlocab) [reply Formula/d/doas.rb:20]: > Would there be a better place to indicate it?

Yes, the upstream documentation. `caveats` are for Homebrew-specific messages. (There are some formulae that don't follow this, but we try to enforce this strictly for new `caveats`.)

**Comment by carlocab** [comment]
> Yes, it does require setuid root but not sure how to achieve it via `brew`.

Doesn't seem like a good fit for Homebrew then TBH. Sorry!

File: Formula/d/doas.rb (hunk 4 above)
[the remark this answers is not in the document]
Comment (Neved4) [reply Formula/d/doas.rb:20]: Understood, some users might miss it in installation but overall it makes sense to me.

**Comment by Neved4** [comment]
@carlocab No worries, thx a lot for your effort! 💙 Closing this!

**Pull request closed without merging**